ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AdminNodeInfo.java

```diff
@@ -77,19 +85,25 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     final EnodeURL enode = maybeEnode.get();
 
     final BytesValue nodeId = enode.getNodeId();
-    response.put("enode", enode.toString());
-    response.put("ip", enode.getIpAsString());
+
+    final String ip = getIp(enode);
+    final int listeningPort = getListeningPort(enode);
+    final int discoveryPort = getDiscoveryPort(enode);
+
+    response.put("enode", getNodeAsString(enode, ip, listeningPort, discoveryPort).toString());
+    response.put("ip", ip);
+
     if (enode.isListening()) {
-      response.put("listenAddr", enode.getIpAsString() + ":" + enode.getListeningPort().getAsInt());
+      response.put("listenAddr", ip + ":" + listeningPort);
```

Comment:
(optional) I tend to prefer `String.format` to `+` for string building because it's cleaner to extend in the future but up to you :)

Reply:
Done